### PR TITLE
Puppetizeing celiometer

### PR DESCRIPTION
**Pull request by genagain**
This installs a ceilometer client on compute nodes. A compute client class was created in the pull request. All of the variables that go into Ceilometer's configuration can be toggled because of this pull request.

### CODE REVIEW

File: quickstack/manifests/controller_common.pp [hunk 1]
```diff
@@ -393,6 +393,14 @@
     admin_address    => $swift_admin_url
   }
 
+  class { 'ceilometer::keystone::auth': 
+    password         => $quickstack::params::ceilometer_user_password,
+    email            => 'ceilometer@bu.edu',
+    public_url       => $quickstack::params::ceilometer_url,
```
Comment (rahulait) [inline quickstack/manifests/controller_common.pp]: email can come from hiera file as well.

File: ceilometer/manifests/client/compute.pp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+
+#  $enhancers = $::ceilometer::params::client_package_names
+  #$enhancers = ['openstack-ceilometer-compute', 'python-ceilometerclient', 'python-pecan']
+
+#  package { 'openstack-ceilometer-compute':
+#    ensure => $ensure,
+#    provider => 'rpm',
+#    tag    => 'openstack',
+#  }
```
Comment (rahulait) [inline ceilometer/manifests/client/compute.pp]: We can remove the commented stuff.

File: ceilometer/manifests/client/controller.pp [hunk 3]
```diff
@@ -34,7 +34,13 @@
...
   class { 'ceilometer::db' : 
-   database_connection => $quickstack::params::ceilometer_db
+    database_connection => $quickstack::params::ceilometer_db
+  }
+
+  class { 'ceilometer::api':
+    keystone_auth_uri => $quickstack::params::ceilometer_auth_uri,
+    keystone_identity_uri => $quickstack::params::ceilometer_identity_uri,
+    keystone_password     => "welcome1"
   }
```
Comment (rahulait) [inline ceilometer/manifests/client/controller.pp]: You need to remove the passwords!!! - 5

Saw the updated request. Its removed there.

**Comment by rahulait** [comment]
Needs to resolve merge conflits.

It would be good if you can come up with 2-3 commits only. Most of the commit changes suggest that the changes were added and later on removed.

I am not sure what does it mean when you say some of the changes on the controller nodes will be configured manually. This needs to be discussed.

**Comment by genagain** [comment]
Because there are too many commits on this branch, I created a new pull request. That way I don't have to resolve any merge conflicts as well. We will discuss how the controller node will be configured

https://github.com/CCI-MOC/kilo-puppet/pull/58

**Pull request closed without merging**